handlers/kinetic_request_ce_attachment_copy_v2/handler/init.rb

```diff
@@ -0,0 +1,192 @@
+      server = @info_values['api_server']+"/"+space_slug
+    else
+      server = @info_values['api_server']
+    end
+
+    user            = @info_values["api_username"]
+    pass            = @info_values["api_password"]
+    error_handling  = @parameters["error_handling"]
+    kapp_slug       = @parameters["kapp_slug"]
+    form_slug       = @parameters["form_slug"]
+    submission_id        = @parameters["submission_id"]
+    to_submission_id        = @parameters["to_submission_id"]
+    to_field_name        = @parameters["to_field_name"]
+    field_name      = @parameters["field_name"]
+    imported_files = []
+    imported_file_details = []
+
+     # Submission API Route including Values
+      puts submission_api_route = "#{server}/app/api/v1/submissions/#{URI.escape(submission_id)}/?include=values"
+      puts("Core submission API Route: \n#{submission_api_route}") if @debug_logging_enabled  
+      # Retrieve the Submission Values
+      submission_result = RestClient::Resource.new(
+        submission_api_route,
+        user: user,
+        password: pass
+      ).get
+
+      puts "Got from submission: #{submission_api_route}"
+      # If the submission exists
+      unless submission_result.nil?
+        submission = JSON.parse(submission_result)["submission"]
+        field_value = submission["values"][field_name]
+        # If the attachment field value exists
+        unless field_value.nil?
+          files = []
+          # Attachment field values are stored as arrays, one map for each file attachment
+          field_value.each_index do |index|
+            file_info = field_value[index]
+            tmp_file_name = File.join(@tmp_dir, file_info['name'])
+            # The attachment file name is stored in the 'name' property
+            # API route to get the generated attachment download link from Kinetic Request CE.
+            attachment_download_api_route = server +
+              '/app/api/v1' +
+              '/submissions/' + URI.escape(submission_id) +
+              '/files/' + URI.escape(field_name) +
+              '/' + index.to_s +
+              '/' + URI.escape(file_info['name']) +
+              '/url'
+            
+            puts "Getting attachment from submission: #{file_info['name']} from field #{field_name}" if @enable_debug_logging
+            
+            # Download file
+            attachment_download_result = RestClient::Resource.new(
+              attachment_download_api_route,
+              user: user,
+              password: pass
+            ).get
+
+            unless attachment_download_result.nil?
+                # get the filehub url to download the file
+                fileUrl = JSON.parse(attachment_download_result)['url']
+                puts "Downloading file: #{file_info['name']} from #{fileUrl}" if @enable_debug_logging
+
+                puts "Downloading file to memory" if @enable_debug_logging
+                file_content = RestClient::Resource.new(
```

Comment:
Again I think this will lead to problems. The file download should be streamed instead of loaded completely into memory.